Add a `CITATION.cff` file

olebole opened the pull request:
### Description

As discussed in #14755, this PR adds a `CITATION.cff` file which lists the way to cite astropy.

Fixes #14755 

There are four major (citeable) parts:
 * the software itself
 * the preferred (5.0) paper
 * the two older papers (0.2 and 2.0)

The file is rather large because it lists all authors of all papers to cite.  The author list of the software was taken from the 5.0 paper, but sorted alphabetically (5.0 has a specific order, with the writers of the paper on first place etc.).

The ORCID ids of the 0.2 paper were filled manually; mainly from the other papers.

Names and affiliations were taken from the `.tex` files, with manual correction of non-ASCII character.

This is accompanied with an extra workflow, `CFF-test.yml` which validates the `CITATION.cff` if it was changed. This is optional, but is the reason why the PR is not "[skip-ci]".

This file must be updated
 * if there is a new paper: change the **preferred-citation** into an entry of the **references** list and create a new **preferred-citation**
 * if there is a new version:
   - update version number, DOI, date
   - check and update authors list

pllim commented on .github/workflows/CFF-test.yml:
```diff
@@ -0,0 +1,21 @@
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: citation-file-format/cffconvert-github-action@2.0.0
```
Interesting... Can we be sure that this Action will be maintained? Which company/institution manages this?

kevinmatthes replied:
The Action is Docker image of the corresponding Python 3 CLI `cffconvert`.  The team behind the CLI also maintains the Action and researches on release automation, at the moment, such that each new release of the CLI would also update and release the Action automatically.  The CLI is expected to receive updates when the CFF version is updated.  Until then, there should be no changes to both the CLI and the Action.

pllim replied:
Thank you for the clarifications! 😸

pllim commented:
pre-commit.ci autofix

pllim commented:
Ugh... the autofix is actually wrong. Sorry. You would need to drop the bot commit. 😢 

@nstarman , how do we prevent the bot from doing this?

kevinmatthes commented:
> Should we add instructions on how to update this file in the future in the dev docs somewhere?

There is also an [Action](https://github.com/kevinmatthes/cff-release-today) to automatically update the release date in the CFF on release, you could plug it into already existing release workflows.  For the version information update, there are several version number incrementing systems like `bump2version` which are known to work very well for CFF.

olebole commented:
One change that would make it easier to maintain is to remove the complete authors list from the main, "software" part and leave there only the generic "Astropy developers" section. This would remove the need to update the list for every new version. The author lists of the papers is static and doesn't need maintainance.
Having an up-to-date software author's list is a bit of a challenge; specifically when there is no programmatic implementation of the rules who counts as author. I would pragmatically propose to leave out individual authors here unless we have such an implementation (that then could be run in an Github action).

pllim commented:
I looks like your branch was built on an older version of `main`. Can you please rebase to pick up CI fixes?

```
git fetch upstream main
git rebase upstream/main
```

Also, `+1,587 −0` is hard to review for me. How do I make this CFF file human-readable?

olebole commented:
I rebased the change. For human readability: let me play a bit with the Python packages provided by https://citation-file-format.github.io/. 

@kevinmatthes maybe you know whether is there already a tool that completely renders a CFF file? A quick check with [cffconvert](https://pypi.org/project/cffconvert) gave me the impression that it converts only the main (software) citation, not the **preferred-citation** or the **references**. I agree that a 1.5k lines file is hard to review.

nstarman commented:
> Ugh... the autofix is actually wrong. Sorry. You would need to drop the bot commit. 😢
> 
> @nstarman , how do we prevent the bot from doing this?

We can hide the cff file from code spell by adding an "exclude" command to its pre-commit config at https://github.com/astropy/astropy/blob/e62fd731c313a839fcd1fc257d7c0cd8c0af0eb9/.pre-commit-config.yaml#L57-L63

olebole commented:
@pllim Here comes a poor-mans-rendering of the content. I hope this helps to review. The code that renders it is attached for reference. 

[render_cff.py.txt](https://github.com/astropy/astropy/files/11635952/render_cff.py.txt)


---

# Astropy

* **Message:** If you use this software, please cite it using the metadata from this file.
* **Repository (code):** https://github.com/astropy/astropy
* **URL:** https://www.astropy.org
* **License:** BSD-3-Clause
* **Version:** v5.3
* **Release date:** 2023-05-23
* **identifiers**
  - doi: DOI for version 5.3 10.5281/zenodo.7963327


## Authors

* Astropy Collaboration


## Preferred citation

### The Astropy Project: Sustaining and Growing a Community-oriented Open-source Project and the Latest Major Release (v5.0) of the Core Package

#### Authors

* Astropy Collaboration
* [Adrian M. Price-Whelan](https://orcid.org/0000-0003-0872-7098), Center for Computational Astrophysics, Flatiron Institute, 162 5th Ave, New York, NY 10010, USA
* [Pey Lian Lim](https://orcid.org/0000-0003-0079-4114), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Nicholas Earl](https://orcid.org/0000-0003-1714-7415), Department of Astronomy, University of Illinois at Urbana-Champaign, 1002 W. Green St., Urbana, IL 61801 USA
* [Nathaniel Starkman](https://orcid.org/0000-0003-3954-3291), David A. Dunlap Department of Astronomy & Astrophysics, University of Toronto, 50 St. George Street, Toronto ON M5S 3H4, Canada
* [Larry Bradley](https://orcid.org/0000-0002-7908-9284), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [David L. Shupe](https://orcid.org/0000-0003-4401-0430), IPAC, MC 100-22, Caltech, 1200E. California Blvd. Pasadena, CA 91125
* [Aarya A. Patil](https://orcid.org/0000-0002-7626-506X), David A. Dunlap Department of Astronomy & Astrophysics, University of Toronto, 50 St. George Street, Toronto ON M5S 3H4, Canada
* [Lia Corrales](https://orcid.org/0000-0002-5466-3817), University of Michigan Astronomy, 1085 S. University, Ann Arbor MI 48109, United States
* [C. E. Brasseur](https://orcid.org/0000-0002-9314-960X), School of Physics and Astronomy, University of St Andrews, North Haugh, St Andrews KY16 9SS, Scotland
* [Maximilian Nöthe](https://orcid.org/0000-0001-7993-8189), Astroparticle Physics, TU Dortmund University, Otto-Hahn-Str. 4a, 44227 Dortmund, Germany
* [Axel Donath](https://orcid.org/0000-0003-4568-7005), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Erik Tollerud](https://orcid.org/0000-0002-9599-310X), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Brett M. Morris](https://orcid.org/0000-0003-2528-3409), Center for Space and Habitability, University of Bern, Gesellschaftsstrasse 6, 3012 Bern, Switzerland
* [Adam Ginsburg](https://orcid.org/0000-0001-6431-9633), Department of Astronomy, University of Florida, P.O. Box 112055, Gainesville, FL, USA
* Eero Vaher, Lund Observatory, Department of Astronomy and Theoretical Physics, Lund University, Box 43, 22100 Lund, Sweden
* Benjamin A. Weaver, NSF's National Optical-Infrared Astronomy Research Laboratory, 950 North Cherry Avenue, Tucson, AZ 85719, United States
* [James Tocknell](https://orcid.org/0000-0001-6637-6922), School of Mathematical and Physical Sciences, Macquarie University, Sydney, NSW 2109, Australia
* [William Jamieson](https://orcid.org/0000-0001-5976-4492), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Marten H. van Kerkwijk](https://orcid.org/0000-0002-5830-8505), David A. Dunlap Department of Astronomy & Astrophysics, University of Toronto, 50 St. George Street, Toronto ON M5S 3H4, Canada
* [Thomas P. Robitaille](https://orcid.org/0000-0002-8642-1329), Aperio Software Ltd., Insight House, Riverside Business Park, Stoney Common Road, Stansted, Essex, CM24 8PL, UK
* Bruce Merry, South African Radio Astronomy Observatory, 2 Fir Street, Black River Park, Observatory 7925, South Africa
* Matteo Bachetti, Istituto Nazionale di Astrofisica/Osservatorio Astronomico di Cagliari, via della Scienza 5, I-09047 Selargius (CA), Italy
* [H. Moritz Günther](https://orcid.org/0000-0003-4243-2840), MIT Kavli Institute for Astrophysics and Space Research, 77 Massachusetts Avenue, Cambridge, MA 02139, USA
* Thomas L. Aldcroft, Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Jaime A. Alvarado-Montes](https://orcid.org/0000-0003-0353-9741), School of Mathematical and Physical Sciences, Macquarie University, Sydney, NSW 2109, Australia
* [Anne M. Archibald](https://orcid.org/0000-0003-0638-3340), Newcastle University
* [Attila Bódi](https://orcid.org/0000-0002-8585-4544), Konkoly Observatory, Research Centre for Astronomy and Earth Sciences (ELKH), H-1121 Budapest, Konkoly Thege Miklós út 15-17, Hungary
* [Shreyas Bapat](https://orcid.org/0000-0002-0870-4665), School of Computing and Electrical Engineering, Indian Institute of Technology Mandi, Kamand, Mandi (H.P.), 175005, India
* [Geert Barentsen](https://orcid.org/0000-0002-3306-3484), Bay Area Environmental Research Institute, P.O. Box 25, Moffett Field, CA 94035, USA
* [Juanjo Bazán](https://orcid.org/0000-0001-7699-3983), CIEMAT, Astroparticle physics, Madrid, Spain
* Manish Biswas, Savitribai Phule Pune University, Pune, Maharashtra, India
* [Médéric Boquien](https://orcid.org/0000-0003-0946-6176), Centro de Astronomía (CITEVA), Universidad de Antofagasta, Avenida Angamos 601, Antofagasta, Chile
* [D. J. Burke](https://orcid.org/0000-0003-4428-7835), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Daria Cara](https://orcid.org/0000-0003-3833-1668), Wake Forest University, 1834 Wake Forest Rd, Winston-Salem, NC, 27109, USA
* [Mihai Cara](https://orcid.org/0000-0002-9294-6551), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Kyle E Conroy](https://orcid.org/0000-0002-5442-8550), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Simon Conseil](https://orcid.org/0000-0002-3657-4191), Aix Marseille Univ, CNRS, CNES, LAM, Marseille, France
* [Matthew W. Craig](https://orcid.org/0000-0001-7988-8919), Department of Physics and Astronomy, Minnesota State University Moorhead, Moorhead, MN 56563
* [Robert M. Cross](https://orcid.org/0000-0003-0081-8024), Independent Astropy Contributor
* [Kelle L. Cruz](https://orcid.org/0000-0002-1821-0650), Hunter College, Physics and Astronomy, 695 Park Avenue, New York, NY 10065
* [Francesco D'Eugenio](https://orcid.org/0000-0003-2388-8172), Cavendish Laboratory and Kavli Institute for Cosmology, University of Cambridge, Madingley Rise, Cambridge, CB3 0HA, United Kingdom
* [Nadia Dencheva](https://orcid.org/0000-0002-5686-9632), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Hadrien A. R. Devillepoix](https://orcid.org/0000-0001-9226-1870), School of Earth and Planetary Sciences, Curtin University, Perth WA 6845, Australia
* [Jörg P. Dietrich](https://orcid.org/0000-0002-8134-9591), Arndtstr. 10, 55124 Mainz, Germany
* [Arthur Davis Eigenbrot](https://orcid.org/0000-0003-0810-4368), National Solar Observatory, 3665 Discovery Drive, Boulder CO. 80303
* Thomas Erben, Argelander-Institut für Astronomie (University of Bonn), Auf dem Hügel 71, 53121 Bonn, Germany
* [Leonardo Ferreira](https://orcid.org/0000-0002-8919-079X), Centre for Astronomy and Particle Physics, University of Nottingham, Nottingham, UK
* [Daniel Foreman-Mackey](https://orcid.org/0000-0002-9328-5652), Center for Computational Astrophysics, Flatiron Institute, 162 5th Ave, New York, NY 10010, USA
* Ryan Fox, Independent Astropy Contributor
* [Nabil Freij](https://orcid.org/0000-0002-6253-082X), Bay Area Environmental Research Institute, P.O. Box 25, Moffett Field, CA 94035, USA
* [Suyog Garg](https://orcid.org/0000-0002-2309-9731), Department of Physics, The University of Tokyo, Bunkyo-ku, Tokyo 113-0033, Japan
* [Robel Geda](https://orcid.org/0000-0003-1509-9966), Department of Astrophysical Sciences, 4 Ivy Lane, Princeton University, Princeton, NJ 08544, USA
* Lauren Glattly, ScienceBetter Consulting, LLC, 4 West 101st Street, No.29, New York, NY 10025
* [Yash Gondhalekar](https://orcid.org/0000-0002-6646-4225), BITS Pilani/Computer Science, Sancoale, Goa 403726, India
* [Karl D. Gordon](https://orcid.org/0000-0001-5340-6774), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [David Grant](https://orcid.org/0000-0001-5878-618X), University of Oxford, Department of Physics, Keble Road, Oxford, OX1 3RH, UK
* [Perry Greenfield](https://orcid.org/0000-0003-2269-0551), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Austen M. Groener](https://orcid.org/0000-0002-6508-2938), Drexel University Department of Physics, 816 Disque Hall 32 S. 32nd Street Philadelphia, PA 19104
* Steve Guest, RAL Space, STFC Rutherford Appleton Laboratory, Harwell, Didcot OX11 0QX, UK
* [Sebastian Gurovich](https://orcid.org/0000-0001-5848-4077), Instituto De Astronomía Teórica y Experimental - Observatorio Astronómico Córdoba( IATEOACUNCCONICET), Córdoba, Argentina
* [Rasmus Handberg](https://orcid.org/0000-0001-8725-4502), Stellar Astrophysics Centre, Department of Physics and Astronomy, Aarhus University, Ny Munkegade 120, DK-8000 Aarhus C, Denmark.
* [Akeem Hart](https://orcid.org/0000-0002-8138-1479), Independent Astropy Contributor
* [Zac Hatfield-Dodds](https://orcid.org/0000-0002-8646-8362), Australian National University, Canberra, Australia
* [Derek Homeier](https://orcid.org/0000-0002-8546-9128), Aperio Software Ltd., Insight House, Riverside Business Park, Stoney Common Road, Stansted, Essex, CM24 8PL, UK
* [Griffin Hosseinzadeh](https://orcid.org/0000-0002-0832-2974), Steward Observatory, University of Arizona, 933 North Cherry Avenue, Tucson, AZ 85721-0065, USA
* [Tim Jenness](https://orcid.org/0000-0001-5982-167X), Vera C. Rubin Observatory, 950 N. Cherry Ave, Tucson, AZ 85719
* [Craig K. Jones](https://orcid.org/0000-0002-0629-3006), The Malone Center for Healthcare in Engineering, Johns Hopkins University
* [Prajwel Joseph](https://orcid.org/0000-0003-1409-1903), Indian Institute of Astrophysics, Bangalore 560034, India
* [J. Bryce Kalmbach](https://orcid.org/0000-0002-6825-5283), DIRAC Institute, Department of Astronomy, University of Washington, Box 351580, Seattle, WA 98195
* [Emir Karamehmetoglu](https://orcid.org/0000-0001-6209-838X), Aarhus University, Department of Physics and Astronomy, Ny Munkegade 120, DK-8000 Aarhus C, Denmark
* [ Mikołaj Kałuszyński](https://orcid.org/0000-0003-0109-5833), Nicolaus Copernicus Astronomical Center, Polish Academy of Sciences, Bartycka 18, 00-716 Warsaw, Poland
* [Michael S. P. Kelley](https://orcid.org/0000-0002-6702-7676), Department of Astronomy, University of Maryland, 4296 Stadium Drive, College Park, MD 20742-0001
* [Nicholas Kern](https://orcid.org/0000-0002-8211-1892), Massachusetts Institute of Technology / Dept. of Physics, 182 Memorial Dr, Cambridge, USA
* [Wolfgang E. Kerzendorf](https://orcid.org/0000-0002-0479-7235), Department of Physics & Astronomy/Computational Mathematics, Science, and Engineering, Michigan State University, East Lansing, MI 48824, USA
* [Eric W. Koch](https://orcid.org/0000-0001-9605-780X), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Shankar Kulumani](https://orcid.org/0000-0002-7822-0471), Johns Hopkins University Applied Physics Laboratory, Laurel, MD USA
* [Antony Lee](https://orcid.org/0000-0003-2193-5369), Laboratoire Photonique, Numérique et Nanosciences UMR 5298, Université de Bordeaux, Institut d'Optique, CNRS, 33400 Talence, France
* [Chun Ly](https://orcid.org/0000-0002-4245-2318), Steward Observatory, University of Arizona, 933 North Cherry Avenue, Tucson, AZ 85721-0065, USA
* [Zhiyuan Ma](https://orcid.org/0000-0003-3270-6844), Department of Astronomy, University of Massachusetts - Amherst, 710 North Pleasant Street, Amherst, MA 01003, USA
* [Conor MacBride](https://orcid.org/0000-0002-9901-8723), Astrophysics Research Centre, School of Mathematics and Physics, Queen's University Belfast, Belfast, BT7 1NN, UK
* [Jakob M. Maljaars](https://orcid.org/0000-0001-8100-9514), Science & Technology Corporation, Delft, The Netherlands
* [Demitri Muna](https://orcid.org/0000-0002-1631-4114), Department of Physics and Astronomy, University of Texas at San Antonio, USA
* [N. A. Murphy](https://orcid.org/0000-0001-6628-8033), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Henrik Norman](https://orcid.org/0000-0003-4189-3450), Winter Way, Lerum, Sweden
* [Richard O'Steen](https://orcid.org/0000-0002-2432-8946), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Kyle A. Oman](https://orcid.org/0000-0001-9857-7788), Institute for Computational Cosmology, Durham University, South Road, Durham DH1 3LE, United Kingdom
* Camilla Pacifici, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Sergio Pascual](https://orcid.org/0000-0002-9351-6051), Departamento de Física de la Tierra y Astrofísica, Fac. CC. Físicas, Universidad Complutense de Madrid, E28040, Madrid, Spain
* [J. Pascual-Granado](https://orcid.org/0000-0003-0139-6951), Instituto de Astrofísica de Andalucía (CSIC). Glorieta de la Astronomía s/n. 18008, Granada, Spain
* [Rohit R. Patil](https://orcid.org/0000-0002-5923-4966), Syracuse University, Syracuse, NY, USA
* [Gabriel I Perren](https://orcid.org/0000-0002-1852-9653), Instituto de Astrofísica de La Plata, IALP (CONICET-UNLP), 1900 La Plata, Argentina
* [Timothy E. Pickering](https://orcid.org/0000-0002-9427-5448), MMT Observatory, Tucson, AZ, USA
* Tanuj Rastogi, Independent Astropy Contributor
* [Benjamin R. Roulston](https://orcid.org/0000-0002-9453-7735), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Daniel F Ryan](https://orcid.org/0000-0001-8661-3825), University of Applied Sciences and Arts Northwest Switzerland, Bahnhofstrasse 6, 5210 Windisch, Switzerland
* [Eli S. Rykoff](https://orcid.org/0000-0001-9376-3135), SLAC National Accelerator Laboratory, Menlo Park, CA 94025, USA
* [Jose Sabater](https://orcid.org/0000-0003-1149-6294), STFC UK Astronomy Technology Centre, Royal Observatory, Blackford Hill, Edinburgh, EH9 3HJ, UK
* [Parikshit Sakurikar](https://orcid.org/0000-0002-9523-5640), International Institute of Information Technology, Hyderabad, India
* [Jesús Salgado](https://orcid.org/0000-0002-3680-4364), SKAO Observatory, Jodrell Bank, Cheshire, UK
* [Aniket Sanghi](https://orcid.org/0000-0002-1838-4757), Department of Astronomy, The University of Texas at Austin, 2515 Speedway, C1400, Austin, TX 78712, USA
* [Nicholas Saunders](https://orcid.org/0000-0003-2657-3889), Institute for Astronomy, University of Hawaiʻi at Māaanoa, 2680 Woodlawn Drive, Honolulu, HI 96822, USA
* [Volodymyr Savchenko](https://orcid.org/0000-0001-6353-0808), Department of Astronomy, University of Geneva, Chemin d'Écogia, 16 CH-1290 Versoix, Switzerland
* Ludwig Schwardt, South African Radio Astronomy Observatory, 2 Fir Street, Black River Park, Observatory 7925, South Africa
* Michael Seifert-Eckert, Independent Astropy Contributor
* [Albert Y. Shih](https://orcid.org/0000-0001-6874-2594), NASA Goddard Space Flight Center, 8800 Greenbelt Road, Greenbelt, MD 20771, USA
* Anany Shrey Jain, Independent Astropy Contributor
* [Gyanendra Shukla](https://orcid.org/0000-0001-9268-3803), Maulana Azad National Institute Of Technology, Bhopal, India
* [Jonathan Sick](https://orcid.org/0000-0003-3001-676X), J.Sick Codes Inc., Penetanguishene ON Canada
* [Chris Simpson](https://orcid.org/0000-0001-8589-4055), Gemini Observatory/NSF's NOIRLab, 670 N. A`ohoku Place, Hilo HI 96720
* [Sudheesh Singanamalla](https://orcid.org/0000-0003-2462-7273), Paul G. Allen School of Computer Science and Engineering, University of Washington, Seattle, WA, USA
* [Leo P. Singer](https://orcid.org/0000-0001-9898-5597), Astroparticle Physics Laboratory, NASA Goddard Space Flight Center, Code 661, Greenbelt, MD 20771, USA
* [Jaladh Singhal](https://orcid.org/0000-0002-8310-0829), TARDIS collaboration
* [Manodeep Sinha](https://orcid.org/0000-0002-4845-1228), Centre for Astrophysics & Supercomputing, Swinburne University of Technology, Hawthorn, VIC 3122, Australia
* [Brigitta M. Sipőcz](https://orcid.org/0000-0002-3713-6337), IPAC, MC 100-22, Caltech, 1200E. California Blvd. Pasadena, CA 91125
* [Lee R. Spitler](https://orcid.org/0000-0001-5185-9876), School of Mathematical and Physical Sciences, Macquarie University, Sydney, NSW 2109, Australia
* [David Stansby](https://orcid.org/0000-0002-1365-1908), Mullard Space Science Laboratory, University College London, Holmbury St. Mary, Surrey, RH5 6NT, UK
* [Ole Streicher](https://orcid.org/0000-0001-7751-1843), Leibniz Institute for Astrophysics, An der Sternwarte 16, 14482 Potsdam, Germany
* Jani Šumak, Independent Astropy Contributor
* [John D. Swinbank](https://orcid.org/0000-0001-9445-1846), ASTRON, Oude Hoogeveensedijk 4, 7991 PD Dwingeloo, The Netherlands
* [Dan S. Taranu](https://orcid.org/0000-0001-6268-1882), Department of Astrophysical Sciences, 4 Ivy Lane, Princeton University, Princeton, NJ 08544, USA
* Nikita Tewary, Bharati Vidyapeeth's College of Engineering/ Electronics and Communication Department, A-4 Block, Baba Ramdev Marg, Shiva Enclave, Paschim Vihar, New Delhi, 110063
* [Grant R. Tremblay](https://orcid.org/0000-0002-5445-5401), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Miguel de Val-Borro](https://orcid.org/0000-0002-0455-9384), Planetary Science Institute, 1700 East Fort Lowell, Tucson, AZ 85719
* [Samuel J. Van Kooten](https://orcid.org/0000-0002-4472-8517), Southwest Research Institute, 1050 Walnut St. Ste 300, Boulder CO 80302, USA
* Zlatan Vasović, Faculty of Physics, University of Belgrade, Studentski trg 12-16, Belgrade, Serbia
* [Shresth Verma](https://orcid.org/0000-0003-0370-5471), Atal Bihari Vajpayee Indian institute of Information Technology and Management Gwalior, MP, India
* [José Vinícius de Miranda Cardoso](https://orcid.org/0000-0001-5410-5551), The Hong Kong University of Science and Technology, Department of Electronic and Computer Engineering, Hong Kong SAR China
* [Peter K. G. Williams](https://orcid.org/0000-0003-3734-3587), Center for Astrophysics | Harvard & Smithsonian, 60 Garden Street, Cambridge, MA, 02138, USA
* [Tom J. Wilson](https://orcid.org/0000-0001-6352-9735), School of Physics, University of Exeter, Stocker Road, Exeter, EX4 4QL, UK
* [Benjamin Winkel](https://orcid.org/0000-0001-6999-3635), Max-Planck-Institut für Radioastronomie, Auf dem Hügel 69, 53121 Bonn, Germany
* [W. M. Wood-Vasey](https://orcid.org/0000-0001-7113-1233), Pittsburgh Particle Physics, Astrophysics, and Cosmology Center (PITT PACC). Physics and Astronomy Department, University of Pittsburgh, Pittsburgh, PA 15260, USA
* [Rui Xue](https://orcid.org/0000-0001-7689-9305), National Radio Astronomy Observatory, 520 Edgemont Road, Charlottesville, VA 22903, USA
* [Peter Yoachim](https://orcid.org/0000-0003-2874-6464), Department of Astronomy, University of Washington, Seattle, WA 98195, USA
* [Chen ZHANG](https://orcid.org/0000-0002-9583-263X), Purple Mountain Observatory, CAS, Qixia District, Nanjing 210023, China
* [Andrea Zonca](https://orcid.org/0000-0001-6841-1058), San Diego Supercomputer Center, University of California San Diego, 9500 Gilman Dr, La Jolla, CA 92093, USA
#### Abstract

The Astropy Project supports and fosters the development of open-source and openly developed Python packages that provide commonly needed functionality to the astronomical community. A key element of the Astropy Project is the core package astropy, which serves as the foundation for more specialized projects and packages. In this article, we summarize key features in the core package as of the recent major release, version 5.0, and provide major updates on the Project. We then discuss supporting a broader ecosystem of interoperable packages, including connections with several astronomical observatories and missions. We also revisit the future outlook of the Astropy Project and the current status of Learn Astropy.  We conclude by raising and discussing the current and future challenges facing the Project.

* **The Astrophysical Journal**, volume 935, issue 2, pages 20, year 2022
* **identifiers**
  - doi: The Astrophysical Journal 10.3847/1538-4357/ac7c74
  - doi: arXiv 10.48550/arXiv.2206.14220
  - other: ADS bibcode 2022ApJ...935..167A
  - url: Publisher PDF (open access) https://iopscience.iop.org/article/10.3847/1538-4357/ac7c74/pdf
  - url: ArXiv eprint (open access) https://arxiv.org/pdf/2206.14220.pdf


## Further references

### The Astropy Project: Building an Open-science Project and Status of the v2.0 Core Package

#### Authors

* Astropy Collaboration
* [A. M. Price-Whelan](https://orcid.org/0000-0003-0872-7098), Department of Astrophysical Sciences, Princeton University, Princeton, NJ 08544, USA
* [B. M. Sipőcz](https://orcid.org/0000-0002-3713-6337)
* [H. M. Günther](https://orcid.org/0000-0003-4243-2840), Kavli Institute for Astrophysics and Space Research, Massachusetts Institute of Technology, 70 Vassar St., Cambridge, MA 02139, USA
* [P. L. Lim](https://orcid.org/0000-0003-0079-4114), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [S. M. Crawford](https://orcid.org/0000-0002-8969-5229), South African Astronomical Observatory, PO Box 9, Observatory 7935, Cape Town, South Africa
* [S. Conseil](https://orcid.org/0000-0002-3657-4191), Univ Lyon, Univ Lyon1, Ens de Lyon, CNRS, Centre de Recherche Astrophysique de Lyon UMR5574, F-69230, Saint-Genis-Laval, France
* [D. L. Shupe](https://orcid.org/0000-0003-4401-0430), Caltech/IPAC, 1200 E. California Blvd, Pasadena, CA 91125
* [M. W. Craig](https://orcid.org/0000-0001-7988-8919), Department of Physics and Astronomy, Minnesota State University Moorhead, 1104 7th Ave S, Moorhead, MN 56563
* [N. Dencheva](https://orcid.org/0000-0002-5686-9632), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [A. Ginsburg](https://orcid.org/0000-0001-6431-9633), National Radio Astronomy Observatory, 1003 Lopezville Rd, Socorro, NM 87801
* [J. T. VanderPlas](https://orcid.org/0000-0002-9623-3401), eScience Institute, University of Washington, 3910 15th Ave NE, Seattle, WA 98195, USA
* [L. D. Bradley](https://orcid.org/0000-0002-7908-9284), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [D. Pérez-Suárez](https://orcid.org/0000-0003-0784-6909), University College London/Research IT Services, Gower St, Bloomsbury, London WC1E 6BT, United Kingdom
* [M. de Val-Borro](https://orcid.org/0000-0002-0455-9384), Astrochemistry Laboratory, NASA Goddard Space Flight Center, 8800 Greenbelt Road, Greenbelt, MD 20771, USA
* T. L. Aldcroft, Harvard-Smithsonian Center for Astrophysics, 60 Garden St., Cambridge, MA, 02138, USA
* [K. L. Cruz](https://orcid.org/0000-0002-1821-0650), Department of Physics and Astronomy, Hunter College, City University of New York, 695 Park Avenue, New York, NY 10065
* [T. P. Robitaille](https://orcid.org/0000-0002-8642-1329), Aperio Software Ltd., Headingley Enterprise and Arts Centre, Bennett Road, Leeds, LS6 3HN, United Kingdom
* [E. J. Tollerud](https://orcid.org/0000-0002-9599-310X), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* C. Ardelean, Department of Physics & Astronomy, University of Western Ontario, 1151 Richmond St, London ON N5X4H1 Canada
* [T. Babej](https://orcid.org/0000-0002-8222-3595), Department of Theoretical Physics & Astrophysics, Masaryk University, Kotlarska 2, 61137 Brno, Czech Republic
* [Y. P. Bach](https://orcid.org/0000-0002-2618-1124), Department of Physics and Astronomy, Seoul National University, Gwanak-gu, Seoul 08826, South Korea
* [M. Bachetti](https://orcid.org/0000-0002-4576-9337), INAF-Osservatorio Astronomico di Cagliari, via della Scienza 5, I-09047, Selargius, Italy
* A. V. Bakanov
* [S. P. Bamford](https://orcid.org/0000-0001-7821-7195), School of Physics & Astronomy, University of Nottingham, University Park, Nottingham NG7 2RD, UK
* [G. Barentsen](https://orcid.org/0000-0002-3306-3484), NASA Ames Research Center, Moffett Field, CA 94043, USA
* [P. Barmby](https://orcid.org/0000-0003-2767-0090), Department of Physics & Astronomy, University of Western Ontario, 1151 Richmond St, London ON N5X4H1 Canada
* [A. Baumbach](https://orcid.org/0000-0002-9374-2729), Heidelberg University, Kirchhoff Institut for Physics, Im Neuenheimer Feld 227, 69116 Heidelberg, Germany
* K. L. Berry
* F. Biscani, Max-Planck-Institut für Astronomie, Königstuhl 17, 69117 Heidelberg, Germany
* [M. Boquien](https://orcid.org/0000-0003-0946-6176), Unidad de Astronomía, Fac. Cs. Básicas, Universidad de Antofagasta, Avda. U. de Antofagasta 02800, Antofagasta, Chile
* K. A. Bostroem, Department of Physics, UC Davis, 1 Shields Ave, Davis, CA, 95616, USA
* L. G. Bouma, Department of Astrophysical Sciences, Princeton University, Princeton, NJ 08544, USA
* [G. B. Brammer](https://orcid.org/0000-0003-2680-005X), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* E. M. Bray
* [H. Breytenbach](https://orcid.org/0000-0001-5391-2386), South African Astronomical Observatory, PO Box 9, Observatory 7935, Cape Town, South Africa
* [H. Buddelmeijer](https://orcid.org/0000-0001-8001-0089), Leiden Observatory, Leiden University, P.O. Box 9513, 2300 RA, Leiden, The Netherlands
* [D. J. Burke](https://orcid.org/0000-0003-4428-7835), Harvard-Smithsonian Center for Astrophysics, 60 Garden St., Cambridge, MA, 02138, USA
* [G. Calderone](https://orcid.org/0000-0002-7738-5389), Istituto Nazionale di Astrofisica, via Tiepolo 11 Trieste, Italy
* [J. L. Cano Rodríguez](https://orcid.org/0000-0002-2187-161X)
* M. Cara, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* J. V. M. Cardoso, Universidade Federal de Campina Grande, Campina Grande, PB 58429-900, Brazil
* S. Cheedella, Department of Physics, Virginia Tech, Blacksburg, VA 24061, USA
* [Y. Copin](https://orcid.org/0000-0002-5317-7518), Université de Lyon, F-69622, Lyon, France; Université de Lyon 1, Villeurbanne; CNRS/IN2P3, Institut de Physique Nucléaire de Lyon
* [L.  Corrales](https://orcid.org/0000-0002-5466-3817), University of Wisconsin - Madison, 475 North Charter Street, Madison, WI 53706
* [D. Crichton](https://orcid.org/0000-0003-1204-3035), Department of Physics and Astronomy, Johns Hopkins University, Baltimore, MD 21218, USA
* D. D'Avella, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [C. Deil](https://orcid.org/0000-0002-4198-4005), Max-Planck-Institut für Kernphysik, PO Box 103980, 69029 Heidelberg, Germany
* [É. Depagne](https://orcid.org/0000-0003-0526-3873), South African Astronomical Observatory, PO Box 9, Observatory 7935, Cape Town, South Africa
* [J. P. Dietrich](https://orcid.org/0000-0002-8134-9591), Faculty of Physics, Ludwig-Maximilians-Universität, Scheinerstr. 1, 81679 Munich, Germany
* A. Donath, Max-Planck-Institut für Kernphysik, PO Box 103980, 69029 Heidelberg, Germany
* M. Droettboom, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [N. Earl](https://orcid.org/0000-0003-1714-7415), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* T. Erben, Argelander-Institut für Astronomie, Auf dem Hügel 71, 53121 Bonn, Germany
* S. Fabbro
* [L. A. Ferreira](https://orcid.org/0000-0002-8919-079X), Instituto de Matemática Estatística e Física – IMEF, Universidade Federal do Rio Grande – FURG, Rio Grande, RS 96203-900, Brazil
* T. Finethy
* [R. T. Fox](https://orcid.org/0000-0003-4291-1091)
* [L. H. Garrison](https://orcid.org/0000-0002-9853-5673), Harvard-Smithsonian Center for Astrophysics, 60 Garden St., Cambridge, MA, 02138, USA
* S. L. J. Gibbons, Institute of Astronomy, University of Cambridge, Madingley Road, Cambridge, CB3 0HA, UK
* D. A. Goldstein, Department of Astronomy, UC Berkeley, 501 Campbell Hall 3411, Berkeley, CA 94720, USA
* [R. Gommers](https://orcid.org/0000-0002-0300-3333), Scion, Private Bag 3020, Rotorua, New Zealand
* [J. P. Greco](https://orcid.org/0000-0003-4970-2874), Department of Astrophysical Sciences, Princeton University, Princeton, NJ 08544, USA
* P. Greenfield, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [A. M. Groener](https://orcid.org/0000-0002-6508-2938), Drexel University, Physics Department, Philadelphia, PA 19104, USA
* F. Grollier
* [A. Hagen](https://orcid.org/0000-0003-2031-7737), Vizual.ai, 3600 O'Donnell St, Suite 250, Baltimore, MD 21224
* P. Hirst, Gemini Observatory, 670 N. Aohoku Pl, Hilo, HI 96720, USA
* [D. Homeier](https://orcid.org/0000-0002-8546-9128), Zentrum für Astronomie der Universität Heidelberg, Landessternwarte, Königstuhl 12, 69117 Heidelberg, Germany
* [A. J. Horton](https://orcid.org/0000-0002-4600-7852), Australian Astronomical Observatory, 105 Delhi Road, North Ryde NSW 2113, Australia
* [G. Hosseinzadeh](https://orcid.org/0000-0002-0832-2974), Las Cumbres Observatory, 6740 Cortona Drive, Suite 102, Goleta, CA 93117-5575, USA
* L. Hu, Imperial College London,  Kensington, London SW7 2AZ, United Kingdom
* [J. S. Hunkeler](https://orcid.org/0000-0003-4989-0289), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [Ž. Ivezić](https://orcid.org/0000-0001-5250-2633), Department of Astronomy, University of Washington, Seattle, WA 98155
* A. Jain, BITS PILANI/Computer Science, Pilani Campus, Rajasthan, India
* [T. Jenness](https://orcid.org/0000-0001-5982-167X), Large Synoptic Survey Telescope, 950 N. Cherry Ave., Tucson, AZ, 85719, USA
* G. Kanarek, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [S. Kendrew](https://orcid.org/0000-0002-7612-0469), European Space Agency, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [N. S. Kern](https://orcid.org/0000-0002-8211-1892), Department of Astronomy, UC Berkeley, 501 Campbell Hall 3411, Berkeley, CA 94720, USA
* [W. E. Kerzendorf](https://orcid.org/0000-0002-0479-7235), European Southern Observatory, Karl-Schwarzschild-Straße 2, 85748 Garching bei München, Germany
* A. Khvalko
* J. King, Max-Planck-Institut für Kernphysik, PO Box 103980, 69029 Heidelberg, Germany
* [D. Kirkby](https://orcid.org/0000-0002-8828-5463), Department of Physics and Astronomy, University of California, Irvine, CA 92697, USA
* A. M. Kulkarni, College of Engineering Pune/Department of Computer Engineering and IT, Shivajinagar, Pune 411005, India
* A. Kumar, Delhi Technological University
* [A. Lee](https://orcid.org/0000-0003-2193-5369), Department of Physics, University of Berkeley, Califonia, CA94709, USA
* [D. Lenz](https://orcid.org/0000-0001-5820-475X), Jet Propulsion Laboratory, California Institute of Technology, 4800 Oak Grove Drive, Pasadena, CA 91109, USA
* [S. P. Littlefair](https://orcid.org/0000-0001-7221-855X), Department of Physics & Astronomy, University of Sheffield, Sheffield, S3 7RH, UK
* [Z. Ma](https://orcid.org/0000-0003-3270-6844), Department of Physics and Astronomy, University of Missouri, Columbia, Missouri, 65211, USA
* [D. M. Macleod](https://orcid.org/0000-0002-1395-8694), Cardiff University, Cardiff CF24 3AA, UK
* [M. Mastropietro](https://orcid.org/0000-0002-6324-5713), Department of Physics and Astronomy, Ghent University, Krijgslaan 281, S9, B-9000 Gent, Belgium
* [C. McCully](https://orcid.org/0000-0001-5807-7893), Las Cumbres Observatory, 6740 Cortona Drive, Suite 102, Goleta, CA 93117-5575, USA
* S. Montagnac, Puy-Sainte-Réparade Observatory
* [B. M. Morris](https://orcid.org/0000-0003-2528-3409), Department of Astronomy, University of Washington, Seattle, WA 98155
* M. Mueller
* [S. J. Mumford](https://orcid.org/0000-0003-4217-4642), SP$^{2}$RC, School of Mathematics and Statistics, The University of Sheffield, U.K.
* [D. Muna](https://orcid.org/0000-0002-1631-4114), Center for Cosmology and Astroparticle Physics, The Ohio State University, 191 West Woodruff Avenue, Columbus, OH 43210
* [N. A. Murphy](https://orcid.org/0000-0001-6628-8033), Harvard-Smithsonian Center for Astrophysics, 60 Garden St., Cambridge, MA, 02138, USA
* S. Nelson, Department of Physics and Astronomy, Minnesota State University Moorhead, 1104 7th Ave S, Moorhead, MN 56563
* [G. H. Nguyen](https://orcid.org/0000-0002-1966-3627), VNU-HCMC, University of Natural Sciences/Faculty of IT, 227 Nguyen Van Cu St., Ward 4, District 5, Ho Chi Minh City, Vietnam
* [J. P. Ninan](https://orcid.org/0000-0001-8720-5612), Dept of Astronomy and Astrophysics, Pennsylvania State University, University Park, PA 16802
* M. Nöthe, Experimental Physics 5, TU Dortmund, Otto-Hahn-Str. 4, 44227 Dortmund, Germany
* S. Ogaz, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [S. Oh](https://orcid.org/0000-0001-7790-5308), Department of Astrophysical Sciences, Princeton University, Princeton, NJ 08544, USA
* J. K. Parejko, Department of Astronomy, University of Washington, Seattle, WA 98155
* N. Parley, University of Reading, Whiteknights Campus, Reading RG6 6BX, UK
* [S. Pascual](https://orcid.org/0000-0002-9351-6051), Departamento de Astrofisica, Universidad Complutense de Madrid, Madrid, Spain
* R. Patil
* A. A. Patil, Pune Institute of Computer Technology, Pune 411043, India
* [A. L. Plunkett](https://orcid.org/0000-0002-9912-5705), European Southern Observatory, Av. Alonso de Córdova 3107, Vitacura, Santiago, Chile
* J. X. Prochaska, Astronomy & Astrophysics, UC Santa Cruz, 1156 High St., Santa Cruz, CA 95064 USA
* T. Rastogi
* V. Reddy Janga, Indian Institute of Technology, Mechanical Engineering, Kharagpur, India
* [J. Sabater](https://orcid.org/0000-0003-1149-6294), Institute for Astronomy (IfA), University of Edinburgh, Royal Observatory, Blackford Hill, EH9 3HJ Edinburgh, U.K.
* P. Sakurikar, IIIT-Hyderabad, India
* M. Seifert
* L. E. Sherbert, Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* [H. Sherwood-Taylor](https://orcid.org/0000-0003-0477-6220)
* A. Y. Shih, NASA Goddard Space Flight Center, 8800 Greenbelt Road, Greenbelt, MD 20771, USA
* [J. Sick](https://orcid.org/0000-0003-3001-676X)
* M. T. Silbiger
* [S. Singanamalla](https://orcid.org/0000-0003-2462-7273), Microsoft Research
* [L. P. Singer](https://orcid.org/0000-0001-9898-5597), Astroparticle Physics Laboratory, NASA Goddard Space Flight Center, 8800 Greenbelt Road, Greenbelt, MD 20771, USA
* [P. H. Sladen](https://orcid.org/0000-0003-1585-225X), Zentrum für Astronomie der Universität Heidelberg, Astronomisches Rechen-Institut, Mönchhofstraße 12--14, 69120 Heidelberg, Germany
* K. A. Sooley
* S. Sornarajah
* [O. Streicher](https://orcid.org/0000-0001-7751-1843), Leibniz Institute for Astrophysics Potsdam (AIP), An der Sternwarte 16, 14482 Potsdam, Germany
* [P. Teuben](https://orcid.org/0000-0003-1774-3436), Astronomy Department, University of Maryland, College Park, MD. 20742, USA
* S. W. Thomas, Institute of Astronomy, University of Cambridge, Madingley Road, Cambridge, CB3 0HA, UK
* [G. R. Tremblay](https://orcid.org/0000-0002-5445-5401), Harvard-Smithsonian Center for Astrophysics, 60 Garden St., Cambridge, MA, 02138, USA
* J. E. H. Turner, Gemini Observatory, Casilla 603, La Serena, Chile
* V. Terrón, Institute of Astrophysics of Andalusia (IAA-CSIC), Granada, Spain
* [M. H. van Kerkwijk](https://orcid.org/0000-0002-5830-8505), Department of Astronomy & Astrophysics, University of Toronto, 50 Saint George Street, Toronto, ON M5S 3H4, Canada
* [A. de la Vega](https://orcid.org/0000-0002-6219-5558), Department of Physics and Astronomy, Johns Hopkins University, Baltimore, MD 21218, USA
* [L. L. Watkins](https://orcid.org/0000-0002-1343-134X), Space Telescope Science Institute, 3700 San Martin Dr., Baltimore, MD 21218, USA
* B. A. Weaver, National Optical Astronomy Observatory, 950 N. Cherry Ave., Tucson, AZ 85719, USA
* [J. B. Whitmore](https://orcid.org/0000-0003-4824-2087), Centre for Astrophysics and Supercomputing, Swinburne University of Technology, Hawthorn, VIC 3122, Australia
* [J. Woillez](https://orcid.org/0000-0002-2958-4738), European Southern Observatory, Karl-Schwarzschild-Straße 2, 85748 Garching bei München, Germany
* [V. Zabalza](https://orcid.org/0000-0003-2638-7648)
#### Abstract

The Astropy Project supports and fosters the development of open-source and openly developed Python packages that provide commonly needed functionality to the astronomical community. A key element of the Astropy Project is the core package astropy, which serves as the foundation for more specialized projects and packages. In this article, we provide an overview of the organization of the Astropy project and summarize key features in the core package, as of the recent major release, version 2.0. We then describe the project infrastructure designed to facilitate and support development for a broader ecosystem of interoperable packages. We conclude with a future outlook of planned new features and directions for the broader Astropy Project.

* **The Astrophysical Journal**, volume 156, issue 3, pages 19, year 2018
* **identifiers**
  - doi: The Astrophysical Journal 10.3847/1538-3881/aabc4f
  - doi: arXiv 10.48550/arXiv.1801.02634
  - other: ADS bibcode 2018AJ....156..123A
  - url: Publisher PDF (open access) https://iopscience.iop.org/article/10.3847/1538-3881/aabc4f/pdf
  - url: ArXiv eprint (open access) https://arxiv.org/pdf/1801.02634.pdf
### Astropy: A community Python package for astronomy

#### Authors

* Astropy Collaboration
* [Thomas P. Robitaille](https://orcid.org/0000-0002-8642-1329), Max-Planck-Institut für Astronomie, Königstuhl 17, Heidelberg 69117, Germany
* [Erik J. Tollerud](https://orcid.org/0000-0002-9599-310X), Department of Astronomy, Yale University, P.O. Box 208101, New Haven, CT 06510, USA
* [Perry Greenfield](https://orcid.org/0000-0003-2269-0551), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* Michael Droettboom, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Erik Bray](https://orcid.org/0000-0001-5391-2386), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* Tom Aldcroft, Harvard-Smithsonian Center for Astrophysics, 60 Garden Street, Cambridge, MA, 02138, USA
* Matt Davis, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Adam Ginsburg](https://orcid.org/0000-0001-6431-9633), Center for Astrophysics and Space Astronomy, University of Colorado, Boulder, CO 80309, USA
* [Adrian M. Price-Whelan](https://orcid.org/0000-0003-0872-7098), Department of Astronomy, Columbia University, Pupin Hall, 550W 120th St., New York, NY 10027, USA
* [Wolfgang E. Kerzendorf](https://orcid.org/0000-0002-0479-7235), Department of Astronomy and Astrophysics, University of Toronto, 50 Saint George Street, Toronto, ON M5S3H4, Canada
* [Alexander Conley](https://orcid.org/0000-0002-3556-0716), Center for Astrophysics and Space Astronomy, University of Colorado, Boulder, CO 80309, USA
* [Neil Crighton](https://orcid.org/0000-0002-3080-6975), Max-Planck-Institut für Astronomie, Königstuhl 17, Heidelberg 69117, Germany
* [Kyle Barbary](https://orcid.org/0000-0002-2532-3696), Argonne National Laboratory, High Energy Physics Division, 9700 South Cass Avenue, Argonne, IL 60439, USA
* [Demitri Muna](https://orcid.org/0000-0002-1631-4114), Department of Astronomy, Ohio State University, Columbus, OH 43210, USA
* Henry Ferguson, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* Frédéric Grollier
* Madhura M. Parikh, S.V.National Institute of Technology, Surat., India
* Prasanth H. Nair
* Hans M. Günther, Harvard-Smithsonian Center for Astrophysics, 60 Garden Street, Cambridge, MA, 02138, USA
* [Christoph Deil](https://orcid.org/0000-0002-4198-4005), Max-Planck-Institut für Kernphysik, P.O. Box 103980, 69029 Heidelberg, Germany
* [Julien Woillez](https://orcid.org/0000-0002-2958-4738), European Southern Observatory, Karl-Schwarzschild-Str. 2, 85748, Garching bei München, Germany
* [Simon Conseil](https://orcid.org/0000-0002-3657-4191), Laboratoire d'Astrophysique de Marseille, OAMP, Université Aix-Marseille et CNRS, Marseille, France
* Roban Kramer, ETH Zürich, Institute for Astronomy, Wolfgang-Pauli-Strasse 27, Building HIT, Floor J, CH-8093 Zurich, Switzerland
* James E. H. Turner, Gemini Observatory, Casilla 603, La Serena, Chile
* [Leo Singer](https://orcid.org/0000-0001-9898-5597), LIGO Laboratory, California Institute of Technology, 1200 E. California Blvd., Pasadena, CA, 91125, USA
* [Ryan Fox](https://orcid.org/0000-0003-4291-1091)
* Benjamin A. Weaver, Center for Cosmology and Particle Physics, New York University, New York, NY 10003, USA
* [Victor Zabalza](https://orcid.org/0000-0003-2638-7648), Max-Planck-Institut für Kernphysik, P.O. Box 103980, 69029 Heidelberg, Germany
* Zachary I. Edwards, Department of Physics and Astronomy, Louisiana State University, Nicholson Hall, Baton Rouge, LA 70803, USA
* K. Azalee Bostroem, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [D. J. Burke](https://orcid.org/0000-0003-4428-7835), Harvard-Smithsonian Center for Astrophysics, 60 Garden Street, Cambridge, MA, 02138, USA
* Andrew R. Casey, Research School of Astronomy and Astrophysics, Australian National University, Mount Stromlo Observatory, via Cotter Road, Weston Creek ACT 2611, Australia
* [Steven M. Crawford](https://orcid.org/0000-0002-8969-5229), SAAO, P.O. Box 9, Observatory 7935, Cape Town, South Africa
* [Nadia Dencheva](https://orcid.org/0000-0002-5686-9632), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* Justin Ely, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* [Tim Jenness](https://orcid.org/0000-0001-5982-167X), Joint Astronomy Centre, 660 N. A`ohōkū Place, Hilo, HI 96720, USA
* Kathleen Labrie, Gemini Observatory, 670 N. A`ohōkū Place, Hilo, Hawaii 96720, USA
* [Pey Lian Lim](https://orcid.org/0000-0003-0079-4114), Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* Francesco Pierfederici, Space Telescope Science Institute, 3700 San Martin Drive, Baltimore, MD 21218, USA
* Andrew Pontzen, Oxford Astrophysics, Denys Wilkinson Building, Keble Road, Oxford OX1 3RH, UK
* Andy Ptak, NASA Goddard Space Flight Center, X-ray Astrophysics Lab Code 662, Greenbelt, MD 20771, USA
* Brian Refsdal, Harvard-Smithsonian Center for Astrophysics, 60 Garden Street, Cambridge, MA, 02138, USA
* [Mathieu Servillat](https://orcid.org/0000-0001-5443-4128), Laboratoire AIM, CEA Saclay, Bat. 709, 91191 Gif-sur-Yvette, France
* [Ole Streicher](https://orcid.org/0000-0001-7751-1843), Leibniz-Institut für Astrophysik Potsdam (AIP), An der Sternwarte 16, 14482 Potsdam, Germany
#### Abstract

We present the first public version (v0.2) of the open-source and community-developed Python package, Astropy. This package provides core astronomy-related functionality to the community, including support for domain-specific file formats such as flexible image transport system (FITS) files, Virtual Observatory (VO) tables, and common ASCII table formats, unit and physical quantity conversions, physical constants specific to astronomy, celestial coordinate and time transformations, world coordinate system (WCS) support, generalized containers for representing gridded as well as tabular data, and a framework for cosmological transformations and conversions. Significant functionality is under active development, such as a model fitting framework, VO client and server tools, and aperture and point spread function (PSF) photometry tools. The core development team is actively making additions and enhancements to the current code base, and we encourage anyone interested to participate in the development of future Astropy versions.

* **Astronomy & Astrophysics**, volume 558, pages 9, year 2013
* **identifiers**
  - doi: Astronomy & Astrophysics 10.1051/0004-6361/201322068
  - doi: arXiv 10.48550/arXiv.1307.6212
  - other: ADS bibcode 2013A&A...558A..33A
  - url: Publisher PDF (open access) https://www.aanda.org/articles/aa/pdf/2013/10/aa22068-13.pdf
  - url: ArXiv eprint (open access) https://arxiv.org/pdf/1307.6212.pdf

olebole commented:
@kevinmatthes 
> Which direction do you mean regarding the rendering; rendering from CFF or to CFF? And what is the other format?

Rendering from CFF to other formats, f.e. BibTeX, MarkDown, HTML. For BibTeX, optimally one would be able to select (and create) individual entries for each paper, compatible with [ADS](https://ui.adsabs.harvard.edu/).

I would rather not split the file, as having multiple citation files may be confusing for those who do not work with this data. Instead, we have some options to significantly reduce the volume if needed:

* we could remove the affiliation for the authors
* we could remove the ORCID ids
* we could remove the abstracts.

All these data are easily retrievable from the papers resp. ADS and may be not needed here. For example, in our [human-readable citation file](https://github.com/astropy/astropy/blob/eaf9e177339758b0854124ea034367efa3484e3a/astropy/CITATION), these data is not included as well. However, the main goal of this file is not readability for humans, so it may be worth to keep it despite the volume.

About Unicode: I fixed this (`yaml.dump(…, allow_unicode=True)`), as it improves readability. However, since CFF is a yaml format, both ways are the same, and machines should not stumble upon it.

kevinmatthes commented:
@olebole

There is a table of existing tools for interaction with CFFs; https://github.com/citation-file-format/citation-file-format#tools-to-work-with-citationcff-files-wrench.  However, this table is neither up-to-date as various requests on adding further software are still pending nor do I know every tool of this catalogue.  I could offer you to implement a CFF-to-BibTeX conversion mode into Aeruginous but this may take some time; I expect about two or three months.

The idea regarding the file splitting would be to store the references in a dedicated directory such that visitors would only see the main CFF in the repository root.  Furthermore, if you cite the works, others might be interested in doing so, as well, such that the references would be reusable for a broader audience.

The importance of Unicode support depends on your actual use cases and with which software you would like to process the CFF.  I usually work with Rust whose string types are built upon Unicode, so Rust software can be safely expected to handle the Unicode strings of this CFF properly.  I experienced that other languages also offer and enhance their support for Unicode in strings as its importance is increasing, thus, I assume that in most cases, machines should be able to work with a CFF containing Unicode without problems.

pllim commented:
@olebole , is this ready for final review or are there more comments to be addressed?

olebole commented:
It is ready for review. I adjusted the non-ASCII chars so that they are readable, and did few minor improvements (ASCL id, URLs for ADS-bibcodes). These are also updated in the [gist with the representation](https://gist.github.com/olebole/2ad8fa260d451fbaaf698f164c717c4b)

We still should discuss about how much information we want to have there: ORCID, Affiliations, Abstracts.

adrn commented on CITATION.cff:
```diff
@@ -0,0 +1,1566 @@
+        orcid: https://orcid.org/0000-0003-0872-7098
+        affiliation: Department of Astrophysical Sciences, Princeton University, Princeton,
+          NJ 08544, USA
+      - family-names: Sipőcz
+        given-names: B. M.
+        orcid: https://orcid.org/0000-0002-3713-6337
```
I think @bsipocz's affiliation got dropped here. Based on the published paper, it should read:
```suggestion
      - family-names: Sipőcz
        given-names: B. M.
        orcid: https://orcid.org/0000-0002-3713-6337
        affiliation: Institute of Astronomy, University of Cambridge, Madingley Road, Cambridge, CB3 0HA, UK
```

olebole replied:
That is correct. I took the repository for the affiliation, and [that had](https://github.com/astropy/astropy-v2.0-paper/blob/c31a57c6ffcf6b040131c15b472d85d9ecdbcc3e/author.tex#L25-L26):
```LaTeX
\author[0000-0002-3713-6337]{B. M. Sip\H{o}cz}
\noaffiliation
```
@

adrn reviewed (commented):
Thanks @olebole for taking this on!

Overall looks great. I caught one missing affiliation that should be there -- do you know what the cause of that would be, and whether there are others that were erroneously dropped?

Also: I agree that the process for upgrading this file should be documented somewhere, but I don't know where. Maybe a new page in the dev docs? Do others have thoughts?

olebole commented:
@adrn Because I took the repositories as source, there may be other mismatches as well, sorry for that. 

I am however not sure whether we really need the affiliations: for the main usage of the CFF file ("document how should Astropy be cited"), this is more that what we put into the current `CITATION` file. Having it here (as well as ORCID ids) increases the expectation in this file - f.e. one would like to (re-)create a BibTeX file from it, and (to avoid confusion) have it identical to the ADS one (because that is somehow the canonical one). To me, this sounds difficult; especially when we need to update this with our next paper.

Without individually specified authors for the software itself, the file needs an update mainly for the updated version (version number, DOI); this should go into a "how-to-create-a-new-release" docs (don't know where).

With a new paper, I think the update is straightforward, and the CFF structure is self-explanatory. And it should be obvious that with a new paper, the `CITATION` files need an update. Without the ORCIDs and affiliation, this is just paper title, author list, journal metadata.

nstarman commented:
@olebole this is looking great! IMO it's nice that this has all the citation information, including ORCID and affiliation. The challenging thing of course is confirming all these details and updating them when the next paper is written. On the latter, we might insist that the author information is built from this CFF, which should ensure that the CFF is always up to date.

olebole commented:
OK; this means however that with a new paper the authors list is primarily maintained as a CFF entry (and then just copied to `CITATION.cff`). Hope that someone remembers this when we start the 4th Astropy paper :eyes:

pllim commented:
There is still a matter of pre-commit bot being unreasonable with the changes here: https://github.com/astropy/astropy/pull/14896#issuecomment-1572413944

How do we ask pre-commit bot to be happy without the autofix, @nstarman ?

nstarman commented:
> There is still a matter of pre-commit bot being unreasonable with the changes here: [#14896 (comment)](https://github.com/astropy/astropy/pull/14896#issuecomment-1572413944)
> How do we ask pre-commit bot to be happy without the autofix, @nstarman ?

https://github.com/astropy/astropy/pull/14896#issuecomment-1573774177

We just need to exclude the file in ``.pre-commit-config.yml``

pllim commented:
@nstarman , are you able to help with the ruff stuff here? 🙏

pllim commented:
Thanks, @nstarman !

@adrn , as for there to document it, I am not sure either. How about a new sub-section under https://docs.astropy.org/en/latest/development/index.html ?

eerovaher commented on .pre-commit-config.yaml:
```diff
@@ -60,6 +60,7 @@ repos:
     hooks:
       - id: codespell
         args: ["--write-changes"]
+        exclude: ".*(CITATION.cff)$"
```
Why is the exclusion being done in the `pre-commit` configuration and not at https://github.com/astropy/astropy/blob/302d190c8f3f03337d91b91fb8561d3be85e0999/pyproject.toml#L360-L361

nstarman replied:
We can change it to there. I only advocated for Astropy to adopt a spell checker, someone else did the great job of setting it up. I've never used codespell outside of pre-commit, so it didn't really occur to me to configure it elsewhere.

nstarman replied:
Done. Thanks @eerovaher.

nstarman replied:
On a separate note, we should clean up the skip string. I wonder if it can be made into a list or some such.

pllim commented:
So... is this PR ready for final review and by who?

olebole commented:
I think it is almost ready; maybe need to update the version info to 6.0.
For potential reviewers, I'd think of @nstarman and @adrn as they were involved in the original issue #14755.

adrn commented:
The cff file looks good to me. But I think the open question is still whether we should mention this somewhere in procedures to document that the file should be updated when the next paper is written. The challenge there is that we don't have a place to document procedures or policies for the astropy papers, so maybe we should make an issue about that broader point and get this PR in as is?

pllim commented:
I agree, @adrn . Would https://github.com/astropy/astropy-project/issues be more appropriate for paper related discussions? Thanks!

nstarman reviewed (approved):
Thanks @olebole for doing this!
I think this is great. If any small mistakes (e.g. affiliation) were made people can open an Issue / PR. 
I agree with @adrn that keeping this updated should become part of our internally documented process for writing papers. We probably also want to add some external facing docs with a minimal explanation of the file format and links explaining how to use it.

pllim commented:
I took the liberty to try to fix pre-commit and do a final workflow clean-up. Hope you don't mind! 🤞

olebole commented:
One open question for me is still whether we want to include the version. That is an optional field, which however would need to be kept up-to-date (and therefore should be part of the release procedures).

pllim commented:
> That is an optional field, which however would need to be kept up-to-date (and therefore should be part of the release procedures).

Is there any clear benefit of including this optional version field that would justify the increase in maintenance burden? If not, my vote is to not do it.

olebole commented:
I think, for keeping the release procedure simple, I will remove it,

pllim commented:
@adrn , if you are okay with this PR as-is, can you please approve for the record? Thanks!

adrn reviewed (approved):
Sounds good! @pllim Making an issue in the astropy-project repo about paper procedures would be great -- do you want me to do that or will you?

pllim commented:
> do you want me to do that or will you?

Can you please open that issue? You would have a better idea on how to explain the problem. I would just say "paper thingy". Thanks! 

Also please cross-link back to here. 

Merging this then.

pllim commented:
Thanks, all!

olebole commented:
Thanks to all who participated in the discussion!
